posts/batchcorrection/index.qmd

```diff
@@ -0,0 +1,736 @@
+
+For those in a hurry to 'just see the code', feel free to jump ahead to the [Harmony Analysis](#tldr) section of this post.
+
+# Introduction 
+
+When analyzing multiple samples or 'batches' of data collected under different technologies or sample conditions, we may need to harmonize the data in a way that preserves biological variation while minimizing technical variation.  
+
+For example, suppose I were interested in analyzing multiple breast cancer tissue samples, but these samples were processed in different labs, by different scientists, or with different sample prep protocols. These technical differences could subtly effect gene detection efficiency in ways that may influence or confound an analysis.
+
+To check for presence of batch effects and whether correction is required, we recommend [plotting](#pcplotbatch) the 'batch' on a UMAP plot or in principal component (PC) space.  In the presence of batch effects, we may observe that cells cluster together by batch more distinctly than they do by their biological cell type. 
+
+For dimension reduction and clustering analyses, there is a multitude of existing research on batch correction methodology [@luecken2019current].  The Harmony method [@harmony] has been recommended as a best-in-class approach in several studies which benchmarked different batch correction methodology for scRNAseq [@tran2020benchmark] [@emmanuel2024batch] . Harmony takes as input a low-dimensional cell embedding such as PCA, and projects cells onto a new unsupervised joint embedding that groups cells by biological cell type rather than dataset-specific conditions.  This Harmony embedding can be used as an input to downstream dimension reduction and clustering algorithms like UMAP[@umap] and Leiden[@leiden]. 
+
+For some other downstream analyses, batch effects can be explicitly handled or may be less of a concern.
+In the case of differential expression (DE) analysis, we could account for the technical effects by specifying 'batch' as a covariate in our DE models.
+Supervised or semi-supervised cell typing analyses that are independent of PCA dimension reductions  (i.e., InSituType [@Danaher2022]), may be less affected by batch effects.  If we were to observe batch effects in this setting, one useful approach would be to perform the cell typing separately by batch before aggregating the cell type annotations.
+
+In this post, we demonstrate using Harmony for batch effect correction on two 6k-plex tonsil tissue datasets; one dataset generated with CosMx 1.5, while the other generated with CosMx 2.0.  The recent CosMx 2.0 release offers noticeable increases in sensitivity; while sensitivity increases will vary from sample-to-sample, transcripts per cell often increase 1.5-2x compared to the same sample processed with previous CosMx 1.5 software.  In case we want to analyze a v1.5 and v2.0 dataset in the same analysis, we might consider treating this variable as a batch effect. 
```

Comment:
"The recent CosMx 2.0 release offers noticeable increases in sensitivity; often obtaining 1.5-2x the transcripts compared to the same sample processed with previous CosMx 1.5 software."